cmd/kubeadm/app/cmd/phases/util.go

```diff
@@ -48,3 +64,18 @@ func runCmdPhase(cmdFunc func(outDir string, cfg *kubeadmapi.InitConfiguration)
 		kubeadmutil.CheckErr(err)
 	}
 }
+
+// SetKubernetesVersion gets current Kubeadm version and sets it as Kubernetes version of master configuration if not set.
+func SetKubernetesVersion(client clientset.Interface, cfg *kubeadmapiv1alpha3.InitConfiguration) error {
+	if cfg.KubernetesVersion != "" {
+		return nil
+	}
+
+	kubeadmVer, _, err := upgrade.NewKubeVersionGetter(client, os.Stdout).KubeadmVersion()
+	if err != nil {
+		return err
+	}
+	cfg.KubernetesVersion = kubeadmVer
```

Comment:
Seems like it would be more reasonable to set `cfg.KubernetesVersion` as `fmt.Sprintf("v1.%d.0", constants.MinimumControlPlaneVersion.Minor()+1)` just like below code does?

https://github.com/kubernetes/kubernetes/blob/c6e91c1f0c8e881b93155248e612278d41ee4933/cmd/kubeadm/app/cmd/config.go#L158-L164

Reply:
looking at the possible options here.